mne/time_frequency/tfr.py

```diff
@@ -230,14 +227,17 @@ def _cwt(X, Ws, mode="same", decim=1, use_fft=True):
                 ret = np.convolve(x, W, mode=mode)
 
             # Center and decimate decomposition
-            if mode == "valid":
+            if mode == 'valid':
                 sz = int(abs(W.size - n_times)) + 1
                 offset = (n_times - sz) // 2
                 this_slice = slice(offset // decim.step,
                                    (offset + sz) // decim.step)
                 if use_fft:
                     ret = _centered(ret, sz)
                 tfr[ii, this_slice] = ret[decim]
+            elif mode == 'full' and not use_fft:
+                ret = ret[(W.size // 2):len(ret) - (W.size // 2)]
+                tfr[ii, :] = ret[decim]
```

Comment:
You are missing a `-1` somewhere, probably need a `(W.size - 1) // 2`, as this only gives the original length back for odd-length signals:
```
>>> import numpy as np
>>> x = np.zeros(10)
>>> w = np.zeros(3)
>>> y = np.convolve(x, w)
>>> y[(w.size // 2):len(y) - (w.size // 2)].size
10
>>> w = np.zeros(4)
>>> y = np.convolve(x, w)
>>> y[(w.size // 2):len(y) - (w.size // 2)].size
9
```
If true, please add a test if possible.